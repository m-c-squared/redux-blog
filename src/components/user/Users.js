import React from 'react';
import PropTypes from 'prop-types';
// import User from './User';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/routes';
import styles from 'styled-components';

const UserLink = styles.ul`
@media (min-width: 374px) {
  display: grid;
    grid-template-columns: 50% 50%;
    grid-row-gap: 30px;
    grid-column-gap: 10px;
    padding: 30px 10px 30px 10px;
}
`;  

const UserList = styles(Link)`
  font-family: 'Roboto', san serif;
  color: rgb(93, 95, 96);
  overflow-wrap: word-break;
`;

const UserTitle = styles.h1`
  margin-top: 1.5em;
  color: rgb(93, 95, 96);
  font-family: 'Roboto', san serif;
`;

const Users = ({ users }) => {
  const userList = users.map(({ id, name }) => 
    <UserList key={id} to={ROUTES.USER_CONTAINER.linkTo(id)}> {name} </UserList>);
  return (
    <>
    <UserTitle> OUR CLIENTS </UserTitle>
    <UserLink>
      {userList}
    </UserLink>
     </>
  );
};

Users.propTypes =  {
  users: PropTypes.array.isRequired
};

export default Users; 
