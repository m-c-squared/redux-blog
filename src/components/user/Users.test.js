import React from 'react';
import renderer from 'react-test-renderer';
import Users from './Users';
import { MemoryRouter } from 'react-router-dom';


describe('users', () => {
  it('matches a snapshot', () => {
    const users = [
      {
        id: '23',
        name:'Clementina DuBuque'
      },
      {
        id: '5',
        name:'Kurtis Weissnat'
      }
    ];
    const tree = renderer.create(
      <MemoryRouter>
        <Users users={users}/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
