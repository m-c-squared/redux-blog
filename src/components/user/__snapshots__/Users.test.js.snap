// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users matches a snapshot 1`] = `
<ul>
  <a
    href="/users/23"
    onClick={[Function]}
  >
     
    Clementina DuBuque
     
  </a>
  <a
    href="/users/5"
    onClick={[Function]}
  >
     
    Kurtis Weissnat
     
  </a>
</ul>
`;
